python/lexer/lexer.py

```diff
@@ -1,36 +1,34 @@
 from tokens import Token, TokenType, char_to_token_type
 
 
-keywords = {
-    'fn' : Token(TokenType.Function, 'fn'),
-    'let': Token(TokenType.Let, 'let')
-}
+keywords = {"fn": Token(TokenType.Function, "fn"), "let": Token(TokenType.Let, "let")}
```

Comment:
this is actually less readable than the removed lines

Reply:
No, PEP8 says your previous version is correct: https://peps.python.org/pep-0008/#indentation. Quote:
> The closing brace/bracket/parenthesis on multiline constructs may either line up under the first non-whitespace character of the last line of list, as in:
```python
my_list = [
    1, 2, 3,
    4, 5, 6,
    ]
result = some_function_that_takes_arguments(
    'a', 'b', 'c',
    'd', 'e', 'f',
    )
```
> or it may be lined up under the first character of the line that starts the multiline construct, as in:
```python
my_list = [
    1, 2, 3,
    4, 5, 6,
]
result = some_function_that_takes_arguments(
    'a', 'b', 'c',
    'd', 'e', 'f',
)
```
Minus the extra whitespace you had before `:`. So PEP8-correct would be this:
```python
keywords = {
    "fn": Token(TokenType.Function, "fn"),
    "let": Token(TokenType.Let, "let")
}
```
That's how I understand it at least.

Reply:
I personally prefer the multi-line as well. Something to consider is reverting this back to a multi-line and adding a trailing comma to make it compliant with PEP8/black.
``` python
keywords = {
    "fn": Token(TokenType.Function, "fn"),
    "let": Token(TokenType.Let, "let"),
}
```

https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#the-magic-trailing-comma